services/addons/images/firmware_manager/upgrader.py

```diff
@@ -33,9 +33,11 @@ def download_blob(self):
         Path(path).mkdir(exist_ok=True)
 
         # Download blob, defaults to GCP blob storage
-        bsp = os.environ.get("BLOB_STORAGE_PROVIDER", "GCP")
-        if bsp == "GCP":
+        bspCaseInsensitive = os.environ.get("BLOB_STORAGE_PROVIDER", "DOCKER").casefold()
+        if bspCaseInsensitive == "gcp":
             download_blob.download_gcp_blob(self.blob_name, self.local_file_name)
+        elif bspCaseInsensitive == "docker":
+            download_blob.download_docker_blob(self.blob_name, self.local_file_name)
         else:
             logging.error("Unsupported blob storage provider")

```

Comment:
Should we throw error to stop any further executions?

Reply:
An exception is now raised when an unsupported blob storage provider is specified.